### PR TITLE
Add two new params for Picard iteration and more tests

### LINKED ISSUE #12939: Add a new input parameter to control Picard iteration
## Rationale

There are cases where a number of Picard iteration is desired but not treated as unconverged. There are also cases where users want only their own custom convergence check and want to avoid the extra not-so-cheap residual evaluations.

## Description

To support these cases, a new input parameter to control Picard iteration can be added, with a name like `disable_picard_residual_norm_check`. Its default value is false, but when it is true, no residual norm will be evaluated and `picard_rel_tol`, `picard_abs_tol` and `picard_force_norms` will be inactive and reaching the maximum number of Picard iterations will be treated as converged. This parameter allows more flexible control on the Picard iteration.

## Impact
A new capability that does not have impact on the current codes.

### PR DESCRIPTION
**Pull request by YaqiWang**
Close #12939.

### CODE REVIEW

**Comment by YaqiWang** [comment]
Looks like this accidentally fixed a bug in this test `multiapps/picard_multilevel.test_force_norms`. Need to look into it later.

**Comment by YaqiWang** [comment]
It is my fault. It will be ready when all tests pass. This change enables user to do fixed number of Picard iterations, could be interesting to @vincentlaboure .

File: framework/src/executioners/PicardSolve.C [hunk 1]
```diff
@@ -136,44 +141,56 @@ PicardSolve::solve()
     {
       if (_has_picard_its)
       {
-        _console << "\n 0 Picard |R| = "
-                 << Console::outputNorm(std::numeric_limits<Real>::max(), _picard_initial_norm)
-                 << '\n';
-
-        for (unsigned int i = 0; i <= _picard_it; ++i)
+        if (_has_picard_norm)
```
Comment (vincentlaboure) [inline framework/src/executioners/PicardSolve.C]: What I was ideally hoping was to have a way to do at most `picard_max_its` iterations before going to the next time step, i.e. go to the next time step if either the tolerances are met or the maximum number of Picard iterations is reached. For pseudo-transients, it makes sense to impose a fixed number of iterations for the 'early' time steps but not when we have almost reached convergence. Here it looks like the tolerances are completely disregarded if _has_picard_norm is true. Maybe you have specific applications in mind?

Comment (YaqiWang) [reply framework/src/executioners/PicardSolve.C]: The parameter I added here is serving duel purpose: 1. avoid residual norm evaluations; 2. not fail the picard solve even maximum of picard iteration is reached. Looks like you want to have two separate parameter for the two goals. Yes, you are right, turning on the current parameter in this PR will disable the residual norm checks completely although it does allow user custom convergence checks.

Comment (YaqiWang) [reply framework/src/executioners/PicardSolve.C]: When you say pseudo-transients, are both the master and subapp pseudo-transient? I guess at least mater is pseudo, right?

Comment (vincentlaboure) [reply framework/src/executioners/PicardSolve.C]: Yes, both are

**Comment by YaqiWang** [comment]
@permcody and @friedmud do you have any comments?

**Comment by permcody** [comment]
I have the same question/concern that Vincent did. I thought we just wanted a single parameter to allow it to proceed when max_its was hit. I'm trying to understand the use case for avoiding the norm. You said you wanted custom criteria, but is that more of a theoretical idea, or something you plan to use? Also, is the norm that expensive that you wish to ignore it? What about the case when your Picard solve would have converged. With the way it stands now, won't you need to keep going to hit the max its?

I'm more inclined to have two separate parameters to allow a solve to proceed when it hasn't reached convergence before hitting max its. If you think you want custom convergence, then perhaps you actually put in a callback for that and have a separate flag?

**Comment by YaqiWang** [comment]
Added a separate param for addressing the comments.

**Comment by YaqiWang** [comment]
This is ready.

**Comment by YaqiWang** [comment]
What are we waiting for? @permcody

**Review by permcody — APPROVED** [review]

**Pull request merged**